docs/_posts/2015-02-24-streamlining-react-elements.md

```diff
@@ -0,0 +1,230 @@
+---
+title: "Streamlining React Elements"
+author: Sebastian Markbåge
+redirect_from: "blog/2015/02/24/streamlining-react-elements.html"
+---
+
+In React 0.13 we’ve added some warnings to some esoteric uses of ReactElement and make them a bit more functional. There is no runtime behavior change in 0.13, only warnings, but we hope to be able to change these behaviors in 0.14 if the community agrees that it is valuable and not too bothersome.
+
+If you use React in an idiomatic way, chances are, you’ll never see any of these warnings. In that case, you can skip this blog post. You can just enjoy the benefits! These changes will unlock simplified semantics, better error messages, stack traces and compiler optimizations!
+
+## Immutable Props
+
+In React 0.12, the props object were mutable. It allows you to do patterns like this:
+
+```js
+var element = <Foo bar={false} />;
+if (shouldUseFoo) {
+  element.props.foo = 10;
+  element.props.bar = true;
+}
+```
+
+The problem is that we don’t have a convenient way to tell when you’re done mutating.
+
+### Problem: Mutating Props You Don’t Own
+
+If you mutate something, you destroy the original value. Therefore, there is nothing to diff against. Imaging something like this:
+
+```js
+var element = this.props.child;
+element.props.count = this.state.count;
+return element;
+```
+
+You take a ReactElement through `props.child` and mutate its property before rendering it. If this component's state update, it won't actually get a new ReactElement in `props.child`. It will be the same one. You're mutating the same props.
```

Comment:
> If this component's state update, ...
"state" is singular, so "updates"

> ... it won't actually get a new ReactElement

"it" is ambiguous here. Since "component's state" is the last noun, it seems like you're referring to that (not the component, which is "you")